### ISSUE TITLE
time: TestLoadFixed failure

### ISSUE THREAD

**Issue by ALTree** [golang/go#17276]
### What version of Go are you using (`go version`)?

```
$ gotip version
go version devel +847bc6d Thu Sep 29 10:56:03 2016 +0200 linux/amd64
$ uname -a
Linux [...] 4.6.0-1-amd64 #1 SMP Debian 4.6.4-1 (2016-07-18) x86_64 GNU/Linux
```
### What operating system and processor architecture are you using (`go env`)?

```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/[...]/gocode"
GORACE=""
GOROOT="/home/[...]/go"
GOTOOLDIR="/home/[...]/go/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build162033413=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
```

### What did you do?
```
$ cd src/time
$ gotip test -run=TestLoadFixed
```
### What did you see?

```
--- FAIL: TestLoadFixed (0.00s)
	time_test.go:947: Now().In(loc).Zone() = "-01", -3600, want "GMT+1", -3600
FAIL
exit status 1
FAIL	time	0.003s
```

**Comment by ALTree** [comment]
A new version of `tzdata`, (`tzdata-2016g`), was released yesterday. From the changelog:

> Several zones in Antarctica and the former Soviet Union, along
     with zones intended for ships at sea that cannot use POSIX TZ
     strings, now use numeric time zone abbreviations instead of
     invented or obsolete alphanumeric abbreviations.  The affected
     zones are [...] **Etc/GMT+1** [...]

**Comment by ALTree** [comment]
Cooking up a patch.

**Comment by ALTree** [comment]
https://go-review.googlesource.com/#/c/29995

**Comment by lfam** [comment]
Applying the fix in c5434f2 to Go 1.4.3 (which can be used as a Go bootstrap compiler) does not work. It fails like this:

```
# Building C bootstrap tool.
cmd/dist

# Building compilers and Go bootstrap tool for host, linux/amd64.
lib9
go tool dist: missing file /tmp/guix-build-go-1.4.3.drv-0/go/include/fmt.h
```

Any advice?

**Comment by ianlancetaylor** [comment]
This issue is closed and the bug you are reporting is not related to it.  Please ask on a forum instead; see https://golang.org/wiki/Questions .  Thanks.

**Comment by banditsmile** [comment]
use go1.4.3 binary package on ubuntu16.04 to build go1.7.3 still face this time package failed problem

**Comment by ianlancetaylor** [comment]
@banditsmile This issue is closed.  It is not expected to be fixed in 1.4.3 or 1.7.3.  If you want to make a case to backport the change to an older release, please open a new issue.  Thanks.

**Comment by gdsotirov** [comment]
The fix from c5434f2 is still not included in just released 1.7.4 and I cannot understand why. I have to manually apply the patch.

**Comment by minux** [comment]
Go 1.7.4 is a security-fix only release. And as this is a test-only
change, it doesn't meet the requirement to be included in a
minor release anyway.

**Comment by ALTree** [comment]
To be fair, the last time this happened the patch was actually cherry-picked in the following minor release.

See: milestone 1.3.3, the issue #8814 was fixes on tip (to be 1.4), but the patch was also included in 1.3.3.

tzdata test failures are harmless, but also pretty annoying because as soon as the major OS receive the updated timezones data, `all.bash` will always fail on **all** architectures and on **all** systems.

**Comment by minux** [comment]
I think Go 1.3 predates the current release guidelines
for minor releases. Please see
https://github.com/golang/go/wiki/Go-Release-Cycle#release-maintenance

"A minor release is issued to address one or more critical problem for which
there is no workaround (typically related to stability or security)."

For this case, obviously the code for package time still works, so people
doesn't even need a workaround.

**Comment by ALTree** [comment]
Ah, thanks. This is the part I missed:

> The only code changes included in the release are the fixes for the specific critical problems

I was convinced it was something like "issued for serious problems, but then it's fine to cherry-pick other small stuff".

**Comment by gdsotirov** [comment]
@minux I haven't omitted the fact that 1.7.4 is a minor security release (there were clear announcements), but still I think cherry picking some other small changes like c5434f2 is worth it. Otherwise, packages like me would have to apply patches as normally most Linux distributions moved tzdata to newer than 2016g and thus it is annoying as @ALTree wrote. Anyway, we'll all have to wait for 1.7.5, 1.8 or whatever else release that includes c5434f2 when it's fully tested (I confirm it's well working on Slackware 14.2).

**Comment by minux** [comment]
If we have set the release guideline formally, then we need to stick to it.

Let me quote from
https://github.com/golang/go/wiki/Go-Release-Cycle#release-maintenance
again, with emphasis mine:

"A minor release is issued to address one or more critical problem for which
there is no workaround (typically related to stability or security). The
only code
changes included in the release are the fixes for the specific critical
problems.
Important documentation-only changes may also be included as well, but
**nothing** more."

The fix for this issue won't be included in any Go 1.7.x minor releases, and
will be include Go 1.8 and above. I hope this is clear.

**Comment by gdsotirov** [comment]
@minux By no way I'm questioning Go's release cycle, but to me it appears too harsh if such small but useful fixes like c5434f2, which help keep Go aligned with the OS updates, are dropped completely from minor releases. And as a distribution packager I'd like to ask what is my _workaround_ for the the time test failure then? I either have to suppress tests completely or I have to manage and apply patches, which as I already wrote is annoying especially for software under active development. Again, I accept Go members' position, but still I'd like express my dissatisfaction. Hopefully, Go 1.8 should be released in a month or so according to the beta1 announcement.

**Comment by bradfitz** [comment]
@gdsotirov, yeah, that's a bit unfortunate. I filed #18163.
